File: tests/benchmarks/prime_bench.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <chrono>
#include <iomanip>

bool is_prime(int num) {
    if (num < 2) return false;
    int limit = static_cast<int>(std::sqrt(num));
    for (int i = 2; i <= limit; ++i) {
        if (num % i == 0) return false;
    }
    return true;
}

int main() {
    int limit = 20000;
    auto start = std::chrono::high_resolution_clock::now();
    int count = 0;
    for (int num = 2; num < limit; ++num) {
        if (is_prime(num)) {
            count++;
        }
    }
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> diff = end - start;
    std::cout << "Result: " << count << std::endl;
    std::cout << "Prime Test: " << std::fixed << std::setprecision(4) << diff.count() << " sec" << std::endl;
    return 0;
}

File: tests/benchmarks/prime_bench.rs
use std::time::Instant;

fn is_prime(num: i32) -> bool {
    if num < 2 { return false; }
    let limit = (num as f64).sqrt() as i32;
    for i in 2..=limit {
        if num % i == 0 { return false; }
    }
    true
}

fn main() {
    let limit = 20000;
    let start = Instant::now();
    let mut count = 0;
    for num in 2..limit {
        if is_prime(num) {
            count += 1;
        }
    }
    let duration = start.elapsed();
    println!("Result: {}", count);
    println!("Prime Test: {:.4} sec", duration.as_secs_f64());
}
